interview-questions.md

```diff
@@ -7,62 +7,69 @@ Then, research the question to expand on your answer. Even if you feel you have
 
+  Researched answer: Indexing is useful for accessing/ modifying its values. When we index into an array we use subscripts. Arrays can be multi-dimensional depending on how many axes they have. I fell down a bit of a rabbithole on this one and don't want to get into too deep talking about a subject I would need more study time to fully comprehend; in the words of Lisa Simpson, "It's better to remain silent and be thought fool than to open your mouth and remove all doubt" - Pretty sure that was Lisa's. For it's purposes I think 'my answer' for this question was probably my most succint. 
 
 
 4. What is iteration?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: As I understand it, to iterate is a way of stating something/ rolling through it (in a manner of speaking that is perhaps not technically correct). A loop, for example, can iterate through an array of numbers checking each one for a certain condition and returning a value based on its criteria. 
 
+  Researched answer: My original answer here was quite far off but I'm not going to edit it. One iteration refers to one execution of a loop. I was reading bits of 'The Self-taught Computer Scientist' on a flight a few weeks ago and if I'd been asked this question then I probably could've given a better answer. It's a really good book on data structures and algorithms. It was discussing Big O notation and the best/ most efficient ways to iterate through large data sets (when time taken becomes more of a key factor in design).
 
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
-  Your answer:
+  Your answer: Pair programming uses a driver and a navigator. The driver is the person physically typing the code and the navigator acts like a co-pilot; collaborating and providing guidance to the driver. Bouncing ideas of each other and working through problems together is an effective way of writing code. 2 heads are better than 1 and, in Aristotle's words, "the whole is greater than the sum of its parts". There's etiquette that should be followed when pair programming but to decent humans most of its common sense and decency- an extension of the 'brave and respectful' space we strive to share - I've very much on board with the LEARN culture. One good rule that I'm glad was highlighted (as I might've been guilty of doing it in haste) was the not writing code while the navigator isn't present rule).
 
-  Researched answer:
+  Researched answer: The driver can also be though of as the 'host' and the navigator as 'the guest'. Part of the guest/ navigator's work is to check and review the code writen by the host/ driver. While the driver focuses more on drafting the code the navigator gives though to the bigger picture allowing for both attention to detail and bigger picture planning.
```

Comment:
These are incredible answers!  Also, really love that you make them lively and sometimes humorous (you know your audience).  Excellent work!